Automattic/woocommerce-payments
# Pass additional data to the intent authentication exception

Slack discussion p1702070481324349/1702053239.912509-slack-C022WMN88KG

Currently, we only throw an exception message `We're not able to process this payment. Please try again later.` and don't have further information for debugging.


#### Changes proposed in this Pull Request
Throw additional data `intent_meta_data` and `order_id` to the intent authentication exception so we have more information for the failed order. 

#### Testing instructions
1. Test with https://github.com/woocommerce/woocommerce/pull/42685
2. Update the [condition](https://github.com/Automattic/woocommerce-payments/blob/0d3bd80bca2f10a6a75b35b01d0f6004bc88436e/includes/class-wc-payment-gateway-wcpay.php#L1273) to `true`
3. Add a log `error_log(print_r($e,true));` behind the [catch](https://github.com/Automattic/woocommerce-payments/blob/0d3bd80bca2f10a6a75b35b01d0f6004bc88436e/includes/class-wc-payment-gateway-wcpay.php#L947) to see the additional data
4. Place an order via WooPay
5. Check the additional data in the debug log

#### If test with WooPay
See the test instruction https://github.com/Automattic/woopay/pull/2394

-------------------


**Post merge**
